utils/common/src/main/java/org/apache/brooklyn/util/JavaGroovyEquivalents.java

```diff
@@ -66,8 +66,9 @@ public static String elvis(String preferred, String fallback) {
     public static String elvisString(Object preferred, Object fallback) {
         return elvis(asString(preferred), asString(fallback));
     }
+    @SuppressWarnings("unchecked")
     public static <T> T elvis(T preferred, T fallback) {
-        return groovyTruth(preferred) ? preferred : fallback;
+        return (T) fix(groovyTruth(preferred) ? preferred : fallback);
```

Comment:
I agree the `fix()` is concerning (returning something different than what is passed in is not expected behaviour in groovy). However, if subsequent use of the value expects to cast it to `java.lang.String`, then we want to convert it from a `GString` to a `String` to let the subsequent java code not throw ClassCastExceptions.

Not quite sure I follow the comment about difference in behaviour. The `elvis(String, String)` will always return a string. The `elvis<T>(string,string)` will return a string, or if passed a `GString` will also return a string.